src/components/Navbar/Searchbar/index.tsx

```diff
@@ -0,0 +1,310 @@
+'use client';
+
+import { setCookie } from '@/lib/actions/client';
+// CSS imports
+import './Searchbar.css';
+
+import Image from 'next/image';
+import { useRouter } from 'next/navigation';
+import { useEffect, useRef, useState } from 'react';
+import { getGlobalSearchData } from '@/lib/actions/get.client';
+import { IGlobalResults } from '@/types/database.interface.misc';
+
+const SearchBar = (props: { className?: string }) => {
+	const [search, setSearch] = useState('');
+	const [searchResults, setSearchResults] = useState<IGlobalResults[]>([]);
+
+	const [isResultsVisible, setIsResultsVisible] = useState(true);
+
+	const ref: any = useRef(null);
+
+	// ref is used to close the search results when clicked outside
+	useEffect(() => {
```

Comment:
You finally were able to achieve without a library ?, Noice

Reply:
![Thank you](https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExaHJ2bzU0dGg5MGVsZWU0NHdzMThscjVvdTNvZ2RlMGlkeWZoY2w1byZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/MZuxgYTprxohb0tpdQ/giphy-downsized-large.gif)